netlink.go

```diff
@@ -67,26 +68,32 @@ func Loopback(ctx context.Context, d Device, size uint64) (idx uint32, wait func
 		return 0, nil, err
 	}
 
+	configErr := make(chan error, 1)
+	defer close(configErr)
+
 	ctx, cancel := context.WithCancel(ctx)
 	ch := make(chan error, 1)
-	go func() {
-		<-ctx.Done()
-		client.Close()
-	}()
 	go func() {
 		err := serve(ctx, serverc, connParameters{exp, defaultBlockSizes})
 		if e := ctx.Err(); e != nil {
 			err = e
 		}
 		cancel()
-		ch <- err
+		client.Close()
 		serverc.Close()
+		if <-configErr == nil {
```

Comment:
This code felt slightly unconventional to me, but the other idea I had of having a `configured bool` that gets set to true (after `Configure()` is called successfully) seemed like it could have a race condition. Definitely open to other ideas here.